src/tests/test-utils.tsx

```diff
@@ -0,0 +1,145 @@
+import {SupportedLocales} from '@open-formulieren/types';
+import {render as rtlRender} from '@testing-library/react';
+import type {RenderOptions, RenderResult} from '@testing-library/react';
+import {Formik} from 'formik';
+import type {FormikErrors, FormikTouched, FormikValues} from 'formik';
+import React from 'react';
+import {IntlProvider} from 'react-intl';
+
+import {BuilderContext} from '@/context';
+import type {ComponentTranslationsRef, DocumentTypeOption, SelectOption} from '@/context';
+import {
+  CONFIDENTIALITY_LEVELS,
+  DEFAULT_AUTH_PLUGINS,
+  DEFAULT_COMPONENT_TREE,
+  DEFAULT_DOCUMENT_TYPES,
+  DEFAULT_FILE_TYPES,
+  DEFAULT_PREFILL_ATTRIBUTES,
+  DEFAULT_PREFILL_PLUGINS,
+  DEFAULT_REGISTRATION_ATTRIBUTES,
+  DEFAULT_VALIDATOR_PLUGINS,
+  sleep,
+} from '@/tests/sharedUtils';
+import {AnyComponentSchema} from '@/types';
+
+import {PrefillAttributeOption, PrefillPluginOption} from '../components/builder/prefill';
+import {RegistrationAttributeOption} from '../components/builder/registration/registration-attribute';
+import {ValidatorOption} from '../components/builder/validate/validator-select';
+
+interface BuilderOptions {
+  supportedLanguageCodes: SupportedLocales[];
+  componentTranslationsRef: ComponentTranslationsRef;
+  defaultComponentTree: AnyComponentSchema[];
+  defaultValidatorPlugins: ValidatorOption[];
+  defaultRegistrationAttributes: RegistrationAttributeOption[];
+  defaultPrefillPlugins: PrefillPluginOption[];
+  defaultPrefillAttributes: {[key: string]: PrefillAttributeOption[]};
+  defaultFileTypes: SelectOption[];
+  defaultdocumentTypes: DocumentTypeOption[];
+  defaultConfidentialityLevels: SelectOption[];
+  registrationAttributesDelay: number;
+}
+
+const contextRender = (
+  ui: React.ReactElement,
+  enableContext: boolean = true,
+  locale: string = 'en',
+  builderOptions: Partial<BuilderOptions> = {},
+  renderOptions: RenderOptions = {}
```

Comment:
discussed in the office, the signature should be updated to `contextRender(ui: React.ReactElement, opts: {...})`